pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -468,6 +474,19 @@ public BrokerService(PulsarService pulsar, EventLoopGroup eventLoopGroup) throws
                         .getBrokerEntryPayloadProcessors(), BrokerService.class.getClassLoader());
 
         this.bundlesQuotas = new BundlesQuotas(pulsar);
+        if (pulsar.getConfiguration().getManagedLedgerDeleteMaxConcurrentRequests() > 0) {
+            log.info("Setting managed ledger deletion max concurrent requests to {} and executor pool size to {}",
+                    pulsar.getConfiguration().getManagedLedgerDeleteMaxConcurrentRequests(),
+                    pulsar.getConfiguration().getManagedLedgerDeleteThreadPoolSize());
+            this.ledgerDeletionSemaphore = new Semaphore(
+                    pulsar.getConfiguration().getManagedLedgerDeleteMaxConcurrentRequests());
+            this.ledgerDeletionExecutorProvider = new ExecutorProvider(
+                    pulsar.getConfiguration().getManagedLedgerDeleteThreadPoolSize(),
+                    "pulsar-ledger-deletion");
```

Comment:
We should shutdown the `ExecutorProvider`

Reply:
fixed.